File: database/src/main/java/org/difft/app/database/models/ForwardContextModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Objects;

@WCDBTableCoding
public class ForwardContextModel {
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    public long databaseId;

    @WCDBField
    public boolean isFromGroup = false;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ForwardContextModel that = (ForwardContextModel) o;
        return isFromGroup == that.isFromGroup;
    }

    @Override
    public int hashCode() {
        return Objects.hash(isFromGroup);
    }
}

File: database/src/main/java/org/difft/app/database/models/ForwardContextModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBTableCoding
import java.util.Objects

@WCDBTableCoding
class ForwardContextModel {
    // Int (not Long): WCDB-KSP generates `== 0` autoincrement guard which does not
    // compile for Long primary keys. Column type stays ColumnType.Integer — no schema
    // change. Local rowid never exceeds 2^31 and is never externally serialized. #901
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    var databaseId: Int = 0

    @WCDBField
    var isFromGroup: Boolean = false

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        other as ForwardContextModel
        return isFromGroup == other.isFromGroup
    }

    override fun hashCode(): Int = Objects.hash(isFromGroup)
}
